### PR TITLE
py-ase: New version 3.18

### PR DESCRIPTION
**Pull request by jabl**
Add ASE version 3.18. ASE 3.18 has dropped support for python 2.x,
hence add a dependency on python@3.5: for it.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-ase/package.py [hunk 1]
```diff
@@ -14,8 +14,11 @@ class PyAse(PythonPackage):
...
 
+    version('3.18.0',
+            sha256='39d45f12def2669605bffc82926acfb13a0d0610e6d82740fa316aafa70f97f9')
     version('3.15.0', '65a0143753517c2df157e53bd29a18e3')
     version('3.13.0', 'e946a0addc5b61e5e2e75857e0f99b89')
 
-    depends_on('python@2.6:')
+    depends_on('python@2.6:', when='@:3.15.0')
+    depends_on('python@3.5:', when='@3.18.0:')
     depends_on('py-numpy', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-ase/package.py]: Several missing dependencies in the latest version, not sure when they were added:
```python
      install_requires=['numpy', 'scipy', 'matplotlib', 'flask'],               
```

File: var/spack/repos/builtin/packages/py-ase/package.py [hunk 2]
```diff
@@ -14,8 +14,11 @@ class PyAse(PythonPackage):
...
     url      = "https://pypi.io/packages/source/a/ase/ase-3.13.0.tar.gz"
 
+    version('3.18.0',
+            sha256='39d45f12def2669605bffc82926acfb13a0d0610e6d82740fa316aafa70f97f9')
     version('3.15.0', '65a0143753517c2df157e53bd29a18e3')
     version('3.13.0', 'e946a0addc5b61e5e2e75857e0f99b89')
 
-    depends_on('python@2.6:')
+    depends_on('python@2.6:', when='@:3.15.0')
+    depends_on('python@3.5:', when='@3.18.0:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-ase/package.py]: I would add `type=('build', 'run')` to these.

Comment (jabl) [reply var/spack/repos/builtin/packages/py-ase/package.py]: Done.

File: var/spack/repos/builtin/packages/py-ase/package.py (hunk 1 above)
Comment (jabl) [reply var/spack/repos/builtin/packages/py-ase/package.py]: Looking at ASE git, 3.14 had these, 3.13 didn't specify any requires. So I guess it's safe to unconditionally require them.

Comment (jabl) [reply var/spack/repos/builtin/packages/py-ase/package.py]: Done.

**Comment by jabl** [comment]
Hmm is there something I can do to fix this coverage check?

**Comment by adamjstewart** [comment]
@jabl see the note at the bottom of https://spack.readthedocs.io/en/latest/contribution_guide.html#coverage

**Pull request merged**